### PR TITLE
OCPBUGS-23358: [release-4.14] Address http2 vulnerability

### PR DESCRIPTION
**Pull request by awgreene**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -278,7 +278,12 @@ github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c/go.mod h1:lNA+9X1NB3Z
 github.com/google/btree v1.0.0/go.mod h1:lNA+9X1NB3Zf8V7Ke586lFgjr2dZNuvo3lPJSGZ5JPQ=
 github.com/google/btree v1.0.1 h1:gK4Kx5IaGY9CD5sPJ36FHiBJ6ZXl0kilRiiCj+jdYp4=
 github.com/google/btree v1.0.1/go.mod h1:xXMiIv4Fb/0kKde4SpL7qlzvu5cMJDRkFDxJfI9uaxA=
+<<<<<<< HEAD
```
Comment (ncdc) [inline go.sum]: Bad 😄

Comment (awgreene) [reply go.sum]: @ncdc what's the proper process to move forward? The cherry-pick creates this merge conflict when the commit is pulled in, should I just take the incoming changes?

Comment (ncdc) [reply go.sum]: You can cherry pick but then you have to resolve the conflict

Comment (awgreene) [reply go.sum]: Got it, I wasn't sure if it had to be an exact representation of the upstream commit, which seemed unachievable. Thanks!

Comment (awgreene) [reply go.sum]: Resolved.

Comment (awgreene) [reply go.sum]: I ran `go mod tidy` to resolve these merge conflicts since they were in `go.sum`.

Comment (awgreene) [reply go.sum]: Or do some kind of <carry> thing for k8s patch bumps?

Comment (awgreene) [reply go.sum]: Resolved offline. Andy and I discussed how we should "stay in the spirit of the commit" if you need to bump different go.mod deps.

**Pull request merged**